### PR TITLE
Using mapping histology groups for transcriptomic-overview plot (PR 4 of 5)

### PR DESCRIPTION
**Pull request by cansavvy**
#### What scientific question is your analysis addressing?

I chunked off some of the original results in this PR into their own PRs to make this one more manageable. See #929 and #928 if you want to know about the `gene-set-enrichment-analysis` changes and `immune-deconv` changes that feed into this. 

This PR finally makes the overall transcriptomic-overview script and plot updated to use `palettes/histology_label_color_table.tsv` table that is created by the notebook added in #899 (but this PR is ultimately dependent on #929 and #928 changes hence why those should be reviewed first).

#### What was your approach?

This PR updated the plot made by `figures/transcriptomic-overview.R` to now use the hex codes and `display_group` for its three panels. 

`transcriptomic-dimension-reduction`
- A quick change to make the function for making the plot use the updated color palette.

- `figures/scripts/transcriptomic-overview.R ` itself needed to have the updated color palette and for it use display group

- And after all the above changes I re-ran `figures/scripts/transcriptomic-overview.R` and pushed the updated results. 

- Lastly, I added a `RUN_LOCAL` option to the generate-figures.sh script so that I'd be able to skip the steps that require more computing power but still generally test that the generate figures script was working after these changes. 

#### What GitHub issue does your pull request address?
Makes more progress toward addressing #898 

### Directions for reviewers. Tell potential reviewers what kind of feedback you are soliciting.
- Is there anything I missed? 

#### Is the analysis in a mature enough form that the resulting figure(s) and/or table(s) are ready for review?
Yes 

### Results
Here's the updated figure:
transcriptomic-overview

#### Reproducibility Checklist

- [x] The dependencies required to run the code in this pull request have been added to the project Dockerfile. -- no new packages are needed. 
- [x] This analysis has been added to continuous integration. 

#### Documentation Checklist

I checked the respective READMEs for these analyses and they did not require any updates for these changes. 

- [x] This analysis module has a `README` and it is up to date.
- [x] This analysis is recorded in the table in `analyses/README.md` and the entry is up to date. 
- [x] The analytical code is documented and contains comments.

### CODE REVIEW

**Comment by cbethell** [comment]
After taking a look at the changes in this PR, I believe it would provide for an easier review if things are broken up into smaller PRs (as per your comment earlier in Slack). 

That said, I believe the main thing would be to break out the `transcriptomic-dimension-reduction` updates (which seems to be responsible for a number of the files changed due to the results files changes) from the `gene-set-enrichment-analysis` updates (which appears to include the most updates made to scripts and also responsible for a number of the files changed)

**Comment by cansavvy** [comment]
Okay. I'll split this up into three small PRs where each PR is a module change.

**Review by cbethell — COMMENTED** [review]
This looks just about ready to go in!

I did checkout your branch and try re-running the `generate-figures.sh` script locally with and without the `RUN_LOCAL=1` option you implemented in this PR and encountered the following error:
```
Reading file data/pbta-snv-strelka2.vep.maf.gz to table 'strelka'.
Traceback (most recent call last):
  File "analyses/snv-callers/scripts/01-setup_db.py", line 291, in <module>
    con.execute(maf_create)
sqlite3.OperationalError: table strelka already exists
```

I am not as familiar with the `snv_callers` module so what may be going on here is not so obvious to me. Have you come across this particular error at any point during your development?

**Comment by cansavvy** [comment]
> I did checkout your branch and try re-running the `generate-figures.sh` script locally with and without the `RUN_LOCAL=1` option you implemented in this PR and encountered the following error:


That is a somewhat expected error if you were trying to run snv-callers (which happens without using `RUN_LOCAL` but if you use this kind of command: `RUN_LOCAL=1 ./generate-figures.sh` you should be able to run it. (It still takes a really long time though even skipping the big steps).

File: analyses/collapse-rnaseq/02-analyze-drops.nb.html [hunk 1]
```diff
@@ -4642,8 +4642,8 @@ <h4>Annotation Table</h4>
...
+<div id="htmlwidget-695d1d7f82b35053562e" style="width:100%;height:auto;" class="datatables html-widget"></div>
```
[the remark this answers is not in the document]
Comment (cansavvy) [reply analyses/collapse-rnaseq/02-analyze-drops.nb.html:4645]: I think its just from a re-run from when it is called in `generate-figures.sh` but I can investigate that to make sure.

[the remark this answers is not in the document]
Comment (cansavvy) [reply analyses/collapse-rnaseq/02-analyze-drops.nb.html:4645]: I'm re-running generate-figures.sh just to get everything square one more time.

**Comment by cansavvy** [comment]
Okey dokey @cbethell it was good you had me run this again because I found an issue with the Telomerase steps (See 631cf46) that had a step that was missing. 

There's a lot of refreshed results files, but none that I don't think are just the results of re-running things that haven't been run in awhile. So let me know what you think. 

Ultimately generate-figures.sh would be better served as a snakemake file so it wouldn't be such a headache for someone to re-run this (i.e. only re-run things that need updating -- something a bash script doesn't know) but for now at least these updates to generate-figures.sh make it so it works. And maybe later we can address making generate-figures.sh into a snakemake run.

**Review by cbethell — COMMENTED** [review]
> There's a lot of refreshed results files, but none that I don't think are just the results of re-running things that haven't been run in awhile.

After taking a look at the most recent changes, I agree with you here @cansavvy.
I just have one minor suggestion re the `figures/Telomerase_Activities.png` plot and that would be to adjust the sizing/positioning as some of the labels on the x-axis in the B panel are now cutoff with the most recent updates. For reference:

Telomerase_Activities

**Comment by cansavvy** [comment]
There's a lot of issues with the telomerase activities plot that were noted in a previous review: https://github.com/AlexsLemonade/OpenPBTA-analysis/pull/927#pullrequestreview-579508629 It's something we know we have to dive back into so I don't want to do too more fiddling with that since it's likely going to have an entire overhaul of it at a later point.

**Review by cbethell — APPROVED** [review]
>There's a lot of issues with the telomerase activities plot that were noted in a previous review: #927 (review) It's something we know we have to dive back into so I don't want to do too more fiddling with that since it's likely going to have an entire overhaul of it at a later point.

Gotcha, in that case, LGTM 👍

**Pull request merged**